pkg/logql/log/parser.go

```diff
@@ -49,10 +53,30 @@ func NewJSONParser() *JSONParser {
 	}
 }
 
+func (j *JSONParser) SetHint(hint string) {
```

Comment:
What is a hint and why is it improving performance? It's not quite clear to me.

Reply:
for example：
logql：
{functionName="feedflow"} | json  | durationMs > 2003
hint is (durationMs> 2003)

simpleJsonParser = (log [ IndexOf(log,"durationMs") , "," ] )

indexOf func is more performant than json func
log
`json
{"level":"info","message":"metrics","functionCostMBMs":193536,"resolveQualifierLatency":0.007786,"isCustomDomainInvoke":true,"serviceLatency":5.307937999999922,"metaLatency":0.001109,"cause":"disableDragonGlobal","scheduleTrace":"0:C2J9JDkvZB18mMZDxAGvhCNjYjQ1MzRiMy1iNDEyLTRmOTYtYWYzNi1hOGVmZDFjYTU1ZWYjZmVlZGZsb3dfX3N0YWJsZSM1ZjFmMWZiYS0yYmQwLTQ5NzAtODA1OS0yNDUzOWVlZmU3ZmEjMDA=","externalServiceVersion":"101","accessKeyID":"","userAgent":"ReactorNetty/0.9.18.RELEASE","numErrors":0,"pickBusyLimit":0,"concurrentCount":27,"functionErrors":0,"durationMs":2099.121799,"forwardedFor":"","timestamp":1644991121,"partitionCount":1,"eeAddress":"33.40.79.69","isHTTPTrigger":false,"requestID":"998b00b1-cd87-4244-862e-f763b0010ebc","fcStatus":200,"concurrentInvocationsPerFunction":27,"functionName":"feedflow","meteringDuration":62846814,"serviceName":"taojimu-fc-prod","traceID":"","method":"POST","network":"none","codeChecksum":"14483028719241002601","eeid":"faas-biz--ecb6be6f-e4d1-4508-a161-1c558348957c_20220216054348.718","mfaPresent":"","preIdleContainer":false,"containerID":"e9d41abe-2215-494a-a474-bdffd705cf8e","invokeDuration":2099121799,"invokeID":"0fb74b04-48ee-46ec-ae24-936939c4201d","queryPathLatency":0.001546,"ip":"33.45.125.83","requestOriginalURI":"/2016-08-15/services/taojimu-fc-prod.feedflow__stable/functions/feedflow/invocations","concurrentInvocations":27,"qualifier":"feedflow__stable","scheduleLatency":0.552185,"maxMemoryUsage":374.21484375,"operation":"InvokeFunction","apiVersion":"2016-08-15","containerReqCount":10,"metaCodeChecksum":"14483028719241002601","host":"feedflow__stable.feedflow.taojimu-fc","errorCode":"","hybridFallBack":true,"codeVersion":"","eaLatency":2103.270315,"concurrentReqLimit":300,"region":"cn-shanghai-corp","module":"api_server","hasFCError":false,"idleFlag":0,"eagleeyeTraceID":"21207bfa16449911210962677efeeb","meteredRespSize":40555,"timestampH":"2022-02-16 05:58:41.100","memoryLimitInMB":3072,"functionTimeoutInSec":50,"runtime":"alinode5","idleLatency":0.474576,"concurrencySoftLimit":0,"enableSlowClientOpt":false,"mode":"sync","accountID":"212068714932184585","isHTTPS":false,"concurrentReqLimitPerFunction":50,"isDarkLaunch":false,"clusterName":"fc-sh-jituan","status":200,"scheduleRetryCount":1,"metaLastModified":"2021-12-21T02:19:39Z","resourceMode":4,"hostName":"function-compute033042074246.ea119","path":"/2016-08-15/services/taojimu-fc-prod.feedflow__stable/functions/feedflow/invocations","latency":2104.429737,"concurrencyLimit":10,"instanceType":"ElasticInstance","invokeException":""}
`

Reply:
Ah. So it's looking ahead. I wonder if a parser Cyril mentioned below plus lazy parsing would be cleaner to implement.

Reply:
Thanks for your review。
I was busy with the coding test of another PR a few days ago (https://github.com/grafana/loki/pull/5455).

Now that PR has been completed, I will try to optimize this PR